src/support/generation.c

```diff
@@ -182,13 +180,15 @@ __gen_drain_callback(
                       session, verbose_orig_level, WT_VERB_CHECKPOINT_PROGRESS, WT_VERBOSE_DEBUG_1);
                 }
                 cookie->verbose_timeout_flags = true;
-                /* Now we have enabled more logs, spin another time to get some information. */
+                /* Now we have enabled more logs, continue waiting to get more information. */
                 continue;
             }
-#endif
+
             if (time_diff_ms >= conn->gen_drain_timeout_ms) {
-                __wt_verbose_error(session, WT_VERB_GENERATION, "%s generation drain timed out",
-                  __gen_name(cookie->base.which));
+                if (counter % 1000 == 0)
```

Comment:
I feel it is better to just log it once; there is no need to log many times of timed-out messages. We are anyway logging the waited time above, so timing out once is enough.